Makefile

```diff
@@ -83,8 +83,10 @@ cross:
 docker: cross
 	cd build && docker build -t $(IMAGE_NAME):$(GIT_HASH) .
 
-release: check test docker
+release-hash: check test docker
 	docker push $(IMAGE_NAME):$(GIT_HASH)
+
+release: check test docker
```

Comment:
No it's not immutable but commit hash are unique for a repository so unless someone is really trying there is no risk that the tag will be overwritten in docker so you can have a high degree of confidence that this is the version that you wanted vs other tags like latest or version numbers that can be easily overwritten by mistakes.